test/integration/targets/one_host/tasks/main.yml

```diff
@@ -0,0 +1,221 @@
+# test code for the one_host module
+
+
+# ENVIRONENT PREPARACTION
+
+- set_fact: test_number= 0
+
+- name: "test_{{test_number}}: copy fixtures to test host"
+  copy:
+    src: testhost/tmp/opennebula-fixtures.json.gz
+    dest: /tmp
+  when:
+    - opennebula_test_fixture
+    - opennebula_test_fixture_replay
+
+
+# SETUP INITIAL TESTING CONDITION
+
+- set_fact: test_number={{ test_number | int + 1 }}
+
+- name: "test_{{test_number}}: ensure the tests hosts are absent"
+  one_host:
+    name: "{{ item }}"
+    state: absent
+    endpoint: "{{ opennebula_endpoint }}"
+    session: "{{ opennebula_session }}"
+    validate_certs: false
+  environment:
+    PYONE_TEST_FIXTURE: "{{ opennebula_test_fixture }}"
+    PYONE_TEST_FIXTURE_FILE: /tmp/opennebula-fixtures.json.gz
+    PYONE_TEST_FIXTURE_REPLAY: "{{ opennebula_test_fixture_replay }}"
+    PYONE_TEST_FIXTURE_UNIT: "test_{{test_number}}_{{ item }}"
+  with_items: "{{opennebula_test.hosts}}"
+  register: result
+
+# NOT EXISTING HOSTS
+
+- set_fact: test_number={{ test_number | int + 1 }}
+
+- name: "test_{{test_number}}: attempt to enable a host that does not exists"
+  one_host:
+    name: badhost
+    state: enabled
```

Comment:
Maybe test all states for a missing host? It should be easy to test it with a loop.

Reply:
Thanks, that is a good idea. Looping different states will make testing easier.

even we can do a transition: enabled, disabled, offline, absent.... 

loads of coverage. will look into it.